monocle/gitlab/mergerequest.py

```diff
@@ -0,0 +1,545 @@
+        }
+        files (first: 100){
+          edges {
+            node {
+              additions
+              deletions
+              path
+            }
+          }
+        }
+        timelineItems (first: 100 itemTypes: [
+              CLOSED_EVENT,
+              PULL_REQUEST_REVIEW,
+              %(force_push_event)s
+              ]) {
+          edges {
+            node {
+              __typename
+              ... on ClosedEvent {
+                id
+                createdAt
+                actor {
+                  login
+                }
+              }
+              ... on PullRequestReview {
+                id
+                createdAt
+                state
+                author {
+                  login
+                }
+              }
+              %(force_push_event_item)s
+            }
+          }
+        }
+        author {
+          login
+        }
+        mergedBy {
+          login
+        }
+        repository {
+          owner {
+            login
+          }
+          name
+        }
+      """  # noqa: E501
+        if include_commits and not self.gql.token_getter.can_read_commit():
+            self.log.info(
+                f"Disable commits fetching for group {self.group} due to unsufficent ACLs"
+            )
+            include_commits = False
+        query_params = {
+            "commits": commits if include_commits else "",
+            "force_push_event": force_push_event if include_commits else "",
+            "force_push_event_item": force_push_event_item
+            if include_commits
+            else "",
+        }
+        return pr_query % query_params
+
+    def _getPage(self, kwargs, prs):
+        qdata = """{
+        repository(owner: "%(group)s", name:"%(repository)s") {
+          pullRequests(
+            first: %(size)s
+            %(after)s
+            orderBy: { field: UPDATED_AT, direction: DESC }
+          ) {
+            totalCount
+            pageInfo {
+              hasNextPage endCursor
+            }
+            edges {
+              node {
+                %(pr_query)s
+              }
+            }
+          }
+        }
+      }"""  # noqa: E501
+        data = self.gql.query(qdata % kwargs)
+        if "data" not in data:
+            self.log.error("No data collected: %s" % data)
+            if "message" in data and "wait a few minutes" in data["message"]:
+                self.log.info("sleeping 2 mn")
+                sleep(120)
+            else:
+                self.log.info("sleeping 20 s")
+                sleep(20)
+            return None
+        if not kwargs["total_prs_count"]:
+            kwargs["total_prs_count"] = data["data"]["repository"][
+                "pullRequests"
+            ]["totalCount"]
+            self.log.info(
+                "Total PRs: %s but will fetch until we reached a PR "
+                "updated at date < %s"
+                % (kwargs["total_prs_count"], kwargs["updated_since"])
```

Comment:
EDIT: seems this is the code standard, at least for this file, so feel free to ignore the comment

isn't this the old string.format method?

not judging and I am not sure if this method is planned to be deprecated (maybe it's just not in my codestyle)

Reply:
It is. Updated it.